Source/WebCore/html/HTMLAttachmentElement.cpp

```diff
@@ -372,6 +372,36 @@ void HTMLAttachmentElement::invalidateRendering()
     }
 }
 
+bool HTMLAttachmentElement::layoutWideLayoutAttachmentOnly() const
+{
+    if (!m_containerElement)
+        return false;
+
+    if (auto* shadowRenderer = m_containerElement->renderer()) {
+        if (shadowRenderer->needsLayout())
+            shadowRenderer->layout();
+        ASSERT(!shadowRenderer->needsLayout());
+        return true;
+    }
+    return false;
+}
+
+bool HTMLAttachmentElement::paintWideLayoutAttachmentOnly(const PaintInfo& paintInfo, const LayoutPoint& offset) const
+{
+    if (!m_containerElement || paintInfo.phase != PaintPhase::Foreground)
+        return false;
+
+    if (auto* shadowRenderer = m_containerElement->renderer()) {
+        PaintInfo shadowPaintInfo = paintInfo;
```

Comment:
This seems like a layering violation. I don't think DOM/HTML should know about painting.

Reply:
Alright, I'll move these to `RenderAttachment`, but I'll have to expose the container element to help.